service/pure/src/pure.rs

```diff
@@ -167,6 +190,29 @@ pub fn get_from(hash: String, external_multiaddr: String) -> IpfsGetResult {
     }
 }
 
+#[marine]
+pub fn dag_get_from(hash: String, external_multiaddr: String) -> IpfsGetResult {
+    log::info!("get from called with hash: {}", hash);
+    let config = load_config();
+    let timeout = config.timeout;
+
+    let particle_id = marine_rs_sdk::get_call_parameters().particle_id;
+    if Multiaddr::from_str(&external_multiaddr).is_err() {
+        return Err(eyre::eyre!("invalid multiaddr: {}", external_multiaddr)).into();
+    }
+
+    let particle_vault_path = format!("/tmp/vault/{}", particle_id);
+    println!("particle_vault_path {}", particle_vault_path);
+    let path = format!("{}/{}", particle_vault_path, hash);
```

Comment:
I think it would be best to move that logic to a separate function to avoid copy pasting. DRY, all that :)